### PR TITLE
feat: api letsql api methods

### PR DESCRIPTION
**Pull request by mesejo**

### CODE REVIEW

File: python/letsql/backends/let/__init__.py [hunk 1]
```diff
@@ -120,6 +120,19 @@ def replace_table(node, _, **_kwargs):
     def do_connect(
         self, config: Mapping[str, str | Path] | SessionContext | None = None
```
Comment (dlovell) [inline python/letsql/backends/let/__init__.py:121]: should we remove `SessionContext` here?

File: python/letsql/config.py [hunk 2]
```diff
@@ -1,22 +1,124 @@
...
 options = Options()
+
+
+def _backend_init():
+    if (backend := options.backend) is not None:
+        return backend
+
+    import letsql
+
+    options.backend = con = letsql.connect()
```
Comment (dlovell) [inline python/letsql/config.py:123]: should there only ever be one instance of `letsql.backends.let.Backend`?

Comment (mesejo) [reply python/letsql/config.py:123]: The user can create multiple instances of `letsql.backends.let.Backend` using the `letsql.connect` method, this instance would be the one used to execute the `api` methods. This is what ibis does, it is also a common pattern in the Python standard library. For example, the modules re and random create class instances and bound the api methods to those intances.

Comment (dlovell) [reply python/letsql/config.py:123]: are we testing using multiple `letsql.backends.let.Backend` instances?

Comment (mesejo) [reply python/letsql/config.py:123]: Done

**Review by dlovell — APPROVED** [review]

**Pull request merged**